### PR TITLE
[BEAM-4318] Enforce ErrorProne analysis in Spark runner project

### PR DESCRIPTION
**Pull request by tengpeng**
This PR follows other PRs that fix the ErrorProne warnings.

Special thanks to @swegner for his help :P

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

### CODE REVIEW

**Comment by tengpeng** [comment]
Retest this please.

**Comment by iemejia** [comment]
Run Spark ValidatesRunner

File: runners/spark/src/test/java/org/apache/beam/runners/spark/metrics/SparkMetricsPusherTest.java [hunk 1]
```diff
@@ -108,7 +111,7 @@ public void processElement(ProcessContext context) {
         counter.inc();
         context.output(context.element());
       } catch (Exception e) {
-        e.printStackTrace();
+        throw new UncheckedIOException((IOException) e);
```
Comment (iemejia) [inline runners/spark/src/test/java/org/apache/beam/runners/spark/metrics/SparkMetricsPusherTest.java]: Can you just LOG.warn here. I don't think that throwing the Exception is the intended use here.

File: runners/spark/src/main/java/org/apache/beam/runners/spark/examples/WordCount.java [hunk 2]
```diff
@@ -54,7 +55,7 @@ public void processElement(ProcessContext c) {
       }
 
       // Split the line into words.
-      String[] words = c.element().split("[^\\p{L}]+");
+      Iterable<String> words = Splitter.onPattern("[^\\p{L}]+").split(c.element());
```
Comment (iemejia) [inline runners/spark/src/main/java/org/apache/beam/runners/spark/examples/WordCount.java]: Is this a error-prone reported issue ? This class is just copied from the examples wordcount one and ideally should not differ from it (not also introduce guava in its dependencies).

Comment (iemejia) [reply runners/spark/src/main/java/org/apache/beam/runners/spark/examples/WordCount.java]: Mmm I se ethis i s clearly a warning on error-prone, can you somehow ignore it.

**Comment by tengpeng** [comment]
@iemejia I have addressed the comments in my latest commit. Thanks!

**Comment by iemejia** [comment]
@tengpeng It seems there are still some issues with checkstyle. Can you please review and squash both commits into one.

**Comment by tengpeng** [comment]
@iemejia My bad. I have passed all style checks and squashed into one commit.

screenshot 2018-05-22 05 42 19

**Comment by swegner** [comment]
This still has [some checkstyle failures](https://scans.gradle.com/s/hwo5gw3npcqee/console-log?task=:beam-runners-spark:checkstyleTest):

```
[ant:checkstyle] [ERROR] /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_GradleBuild/src/runners/spark/src/test/java/org/apache/beam/runners/spark/ProvidedSparkContextTest.java:25: Import com.google.common.collect.ImmutableList appears after other imports that it should precede [ImportOrder]
[ant:checkstyle] [ERROR] /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_GradleBuild/src/runners/spark/src/test/java/org/apache/beam/runners/spark/aggregators/metrics/sink/SparkMetricsSinkTest.java:26: Import com.google.common.collect.ImmutableList appears after other imports that it should precede [ImportOrder]
[ant:checkstyle] [ERROR] /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_GradleBuild/src/runners/spark/src/test/java/org/apache/beam/runners/spark/aggregators/metrics/sink/SparkMetricsSinkTest.java:64: Line is longer than 100 characters (found 126). [LineLength]
```

It looks like your Checkstyle is configured correctly, but maybe you didn't scan the whole project?

**Comment by iemejia** [comment]
Probably it is that he does not have the correct checkstyle rules configured. In any case to ensure that everything is working correctly in your local machine you can run:
```
./gradlew -p runners/spark clean build -Dorg.gradle.caching=false
```
And if everything in the spark module looks good you can push your changes.

**Comment by iemejia** [comment]
@tengpeng any news on this one? It looks pretty close.

**Pull request closed without merging**